### ISSUE TITLE
Small errors in spec for switch

### ISSUE THREAD

**Issue by kmillikin** [dart-lang/sdk#22127]
There are a few small problems with this section that don't seem to have their own issues filed:

1. In the grammar for switchCase, it has
&quot;&quot;&quot;
switchCase:
&nbsp;&nbsp;label\* (case expression ‘:’) statements
;
&quot;&quot;&quot;
While not actually a bug, the parentheses grouping around &quot;case expression ‘:’&quot; suggests that it will be followed by a quantifier like \*, +, or ?.  Since it's not and I can't see any purpose for it, the grouping complicates the production in my opinion.

2. The description several times has thing like
&quot;&quot;&quot;
Given a switch statement of the form
switch (e) {
&nbsp;&nbsp;case label11 . . . label1j1 e1 : s1
&nbsp;&nbsp;. . .
&nbsp;&nbsp;case labeln1 . . . labelnjn en : sn
&nbsp;&nbsp;default: sn+1
}
&quot;&quot;&quot;
The grammar indicates that the labels have to come before the word &quot;case&quot; and that no labels are allowed between &quot;case&quot; and the expression.

3.  There is commentary that says:
&quot;&quot;&quot;
In other words, there is no implicit fall-through between cases. The last case in a switch (default or otherwise) can ‘fall-through’ to the end of the statement.
&quot;&quot;&quot;
But this is misleading.  There is implicit fall-through between empty cases according to the semantics.  A simple fix is to use the wording &quot;no implicit fall-through between non-empty cases&quot;.
